AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -1,31 +1,73 @@
-//
-//  DoubleVideoViewController.swift
-//  AC3.2-MidUnit-6-Exam
-//
-//  Created by Jason Gresh on 2/3/17.
-//  Copyright © 2017 AccessCode. All rights reserved.
-//
+    /*
+ 
+ Use UIImagePickerController to pick the video from the library.
+ Use the AVPlayerItem, AVPlayer, and AVPlayerLayer objects to take the chosen video and display it. Remember,imagePickerController(_:didFinishPickingMediaWithInfo:) will give you a URL for Movies and that can be used to construct an AVPlayerItem.
+ You can determine whether a video is currently playing by checking the rate property of the AVPlayer object. No keys or values need be harmed (or even observed) during the making of this app. (Plain English: No KVO needed).
+ */
 
 import UIKit
+import AVKit
+import MobileCoreServices
+import AVFoundation
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
     var titleForCell: String = "Double Video"
+    var movieURL: URL?
+    var player: AVPlayer?
 
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+      
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
 
     override func viewDidLayoutSubviews() {
         // update your layers' frames here
+        guard let sublayers = self.videoContainerTop.layer.sublayers
+            else {
+                return
+        }
+        
+        for layer in sublayers {
+            layer.frame = self.videoContainerTop.bounds
+        }
     }
     
     @IBAction func loadVideo(_ sender: UIButton) {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.sourceType = .photoLibrary
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
```

Comment:
It was a nice touch to limit choices to only video, as this was not specified but makes sense.